source/includes/markdown/_13-platform-ui.html.md

```diff
@@ -278,11 +271,190 @@ Related References:
+
+When handling requests from Asana, an App Components app should:
+ 
+ - Reject requests with missing or incorrect signatures.
+ - Reject requests with an "expires" time in the past.
+    - **Note**: Make sure to use the correct units for this. "expires" is in milliseconds. If you compare the expiration time to a timestamp in seconds, it will always look like the request expires thousands of years in the future.
+
+If an app uses OAuth for authentication, the app should:
+ 
+ - Prevent OAuth CSRF attacks. This is often done using a one-time CSRF token in the "state" parameter. This can also be done using PKCE instead, if it's supported.
+
+If an app doesn't use OAuth for authentication, the Asana Security Team should manually review the authentication scheme the app uses. In particular, we will try to verify that:
+
+ - An attacker can't authenticate themselves as someone else
+ - An attacker can't force a victim to authenticate as another user (eg. with a CSRF attack)
+
+ <hr>
 
-<span class="beta-indicator">BETA</span> - For access, please see [App Components Beta](/docs/app-components-beta)
+## Authorization
+
+<span class="beta-indicator">BETA</span> - For access, please see [Overview of App Components](/docs/overview-of-app-components)
+
+```html
+<!-- Example: An app might return this HTML in response to the 
+authenticationUrl request after the Oauth flow is completed:
+ -->
+
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>You have successfully connected Asana to the app</title>
+  </head>
+  <body>
+    Success!
+    <script>
+      window.opener.postMessage("success", "ORIGIN");
+      window.close();
+    </script>
+  </body>
+</html>
+```
+
+### Overview
+
+The Asana platform needs confirmation from the App Components app that authentication has completed successfully. When 
+an app is added to a project, the user adding it is sent to the app's `authenticationUrl`. The app may perform OAuth with Asana,
+OAuth with a different app, perform both, or perform none. As long as the app confirms the flow was complete, Asana will
+successfully add the app to the project. This will allow requests to be sent to the app's predefined endpoints.
+
+### How it works
+
+Under the hood, we carry this out by listening for a message from the authentication popup window containing the string "success".
+When we make a request to the app's `authenticationUrl`, the browser opens a popup or "child" window and waits for it to respond
+with "success"  using [window.postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage). The target window
+for this `postMessage` call should be the opener, accessible via
+[window.opener](https://developer.mozilla.org/en-US/docs/Web/API/Window/opener).
+
+Note that for security purposes, the _origin_ included in the event needs to match the `serverUrl` registered to the app. That is,
```

Comment:
FYI that we've aligned on computing the serverUrl from the authenticationUrl, so this should become a lot simpler. I think the form includes a field for the `serverUrl` today, but we could remove it and for manual submissions going forward we could take the protocol + host (URL without the path) of the authenticationUrl and set that as the `serverUrl`

Reply:
Sounds good! Is it okay to remove `serverUrl` from the form now? Server and auth URLs are sometimes one and the same anyways.